pybossa/util.py

```diff
@@ -1156,20 +1159,24 @@ def generate_bsso_account_notification(user):
     return msg
 
 
-def is_annex_response(key, value):
+def check_annex_response(response_value):
     """
-    Recursively check whether a response is a odfoa response. Returns the check
-    result and odfoa response
+    Recursively check whether a response is an odfoa response.
+    If it is a valid odfoa response, return odfoa response data. Otherwise
+    return None
     """
-    odfoa_keys = {"version", "source-uri", "odf", "oa"}
-    if type(value) is dict and all(odfoa_key in value for odfoa_key in odfoa_keys):
-        return True, value
-    if type(value) is dict:
-        for k, v in value.items():
-            is_annex, response = is_annex_response(key, v)
-            if is_annex:
-                return is_annex, response
-    return False, None
+    if type(response_value) is not dict:
+        return None
```

Comment:
Though ```return``` and ```return None``` means the same, ```return``` tends to be used when a function doesn't return anything, like ```void``` in C++. This function, we do need to return a response data. I think we it could be clearer to explicitly return "None", especially in a recursive function.

Reply:
as return or return None is same. its fine if return None makes it more explicit